x-pack/plugins/aiops/public/embeddables/change_point_chart/types.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import type { DefaultEmbeddableApi } from '@kbn/embeddable-plugin/public';
+import type {
+  HasEditCapabilities,
+  PublishesDataViews,
+  PublishesTimeRange,
+  PublishingSubject,
+  SerializedTimeRange,
+  SerializedTitles,
+} from '@kbn/presentation-publishing';
+import type { ChangePointDetectionViewType } from '@kbn/aiops-change-point-detection/constants';
+import type { SelectedChangePoint } from '../../components/change_point_detection/change_point_detection_context';
+
+export type ViewComponent = FC<{
+  changePoints: SelectedChangePoint[];
+  interval: string;
+  onRenderComplete?: () => void;
+}>;
+
+export interface ChangePointComponentApi {
```

Comment:
Nit: every item you put in here is completely exposed to any other panel or action in the page's context. This isn't necessarily bad, but I'd recommend trying to avoid putting internal state management details on the embeddable's API if possible.

This can be accomplished by passing these things down as props, or by passing a separate `stateManager` object.  You can even pass these things to an editor etc internally without having them on the external API.